app/graphql_server/main.go

```diff
@@ -44,11 +48,19 @@ func main() {
 		middleware.NewAuthMiddleware,
 	}
 
+	// Define a Mux
+	router := mux.NewRouter()
+	router.Use(middleware.NewPrometheusMiddleware)
```

Comment:
Can you move the rest of the middleware here?

Can you just add all of the middleware? If so do we even need the chain middleware?

Reply:
It seems we can add middleware via Negroni instead?

https://github.com/urfave/negroni#with

^This seems like the more standard way to middleware that I am seeing (with gorilla used for routing mainly). If you don't mind can we move to this for middleware registration?

Reply:
(If not I can open a new PR after this one to do that move, no worries!)

Reply:
All about it, let me look at these patterns and implement. 

Part of why I'm getting in early is because my company has robust stuff in place, and I wanna fiddle with it from the start 😆 (and build my go dev experience that way)

Reply:
Okay, using Negroni will involve a bit of a rewrite of the handler funcs to match the `negroni.Handler` interface:
```go
type Handler interface {
  ServeHTTP(w http.ResponseWriter, r *http.Request, next http.HandlerFunc)
}```
I'm able to do that on my handlers, but I think doing it on the existing ones in my PR breaks the spirit of atomic PRs and makes roll back really hard. Should I make a second PR to change the existing right first, or do it all in one, do you think?

Reply:
I think doing the rest of the move on a follow up PR sounds fine yeah 👍

Reply:
Sweet. Now just to finish trying to wrangle Gorilla, Negroni, and Prometheus together to play nicely. 🤣

Reply:
Okay, I couldn't get negroni to work properly. We poked at it in #backend-dev a bit as well, but ultimately something about the patterns here didn't "click" for me.